fixbackend/workspaces/dependencies.py

```diff
@@ -35,13 +35,16 @@
 
 async def get_optional_user_workspace(
     workspace_id: Annotated[WorkspaceId, Path()],
-    user: AuthenticatedUser,
+    user: OptionalAuthenticatedUser,
     workspace_repository: WorkspaceRepositoryDependency,
 ) -> Workspace | WorkspaceError:
     workspace = await workspace_repository.get_workspace(workspace_id)
     if workspace is None:
         return "WorkspaceNotFound"
```

Comment:
Consider moving it after the user check - no need to hit the database for unauthorized requests.